crates/l2/sequencer/l1_watcher.rs

```diff
@@ -91,14 +93,31 @@ impl L1Watcher {
                 .into();
         }
 
-        let current_block = self.eth_client.get_block_number().await?;
+        let Some(latest_block_to_check) = self
+            .eth_client
+            .get_block_number()
+            .await?
+            .checked_sub(self.l1_block_delay.into())
+        else {
+            warn!("Too close to genesis to request deposits");
+            return Ok(vec![]);
+        };
 
         debug!(
-            "Current block number: {} ({:#x})",
-            current_block, current_block
+            "Latest possible block number with {} blocks of delay: {} ({:#x})",
+            self.l1_block_delay, latest_block_to_check, latest_block_to_check
         );
 
-        let new_last_block = min(self.last_block_fetched + self.max_block_step, current_block);
+        // latest_block_fetched could be greater than latest_block_to_check if the node is stopped and l1_block_delay is changed
+        if self.last_block_fetched > latest_block_to_check {
+            warn!("Last block fetched is greater than latest trusted block");
```

Comment:
I'd use "safe" instead of trusted, what do you think?
```suggestion
            warn!("Last block fetched is greater than latest safe block");
```

Reply:
Agree done [here](https://github.com/lambdaclass/ethrex/pull/2816/commits/df1e64520d2f6a7aee18802ccd76bef5653b6591)